mmv1/templates/terraform/custom_delete/clear_instance_settings.go.erb

```diff
@@ -0,0 +1,49 @@
+url, err := tpgresource.ReplaceVars(d, config, "{{ComputeBasePath}}projects/{{project}}/zones/{{zone}}/instanceSettings")
+if err != nil {
+        return err
+}
+
+project, err := tpgresource.GetProject(d, config)
+if err != nil {
+        return fmt.Errorf("Error fetching project for InstanceSettings: %s", err)
+}
+
+res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+	Config:    config,
+	Method:    "GET",
+	Project:   project,
+	RawURL:    url,
+	UserAgent: userAgent,
+})
+
+if err != nil {
+		return err
+}
+
+obj := map[string]interface {}{"fingerprint":res["fingerprint"], "metadata":map[string]interface {}{"items":map[string]string{}}}
```

Comment:
As this resource is not truly being created within Terraform -- is it intended for it to be wiped everything to "blank" settings for the Zone if it is removed from Terraform?

Reply:
These are settings which exists by default and hence cannot be removed. The create will add/update some key value pairs in the settings for a particular zone. These metadata key value pair will be accessed by all the VMs in that zone. To simulate the delete functionality, these settings are being cleared to avoid VMs from using them anymore.